.github/workflows/_e2e.yaml

```diff
@@ -103,6 +104,23 @@ jobs:
           if [[ "${{ env.E2E_PARAM_TARGET }}" == "multizone" ]]; then
             export KUMA_DEFAULT_RETRIES=60
           fi
+          
+          function on_exit()
+          {
+            docker logout docker.io
+          }
+          
+          # we pull a few images during the E2E run, sometimes we get rate-limited by docker hub
+          # to prevent this, we support specifying a pull credential here
+          if [[ "${{ secrets.DOCKERHUB_PULL_CREDENTIAL }}" != "" ]]; then
+            export K3D_PULL_CREDENTIAL="${{ secrets.DOCKERHUB_PULL_CREDENTIAL }}"
+            export KIND_PULL_CREDENTIAL="${{ secrets.DOCKERHUB_PULL_CREDENTIAL }}"
```

Comment:
You don't need to differentiate these because either you are K3D or KIND you are never both.